src/main/java/com/cryptomorin/xseries/XItemStack.java

```diff
@@ -749,6 +770,52 @@ public static ItemStack deserialize(@Nonnull ConfigurationSection config) {
         return item;
     }
 
+    /**
+     * Deserialize an ItemStack from a {@code Map}.
+     *
+     * @param serializedItem the map holding the item configurations to deserialize
+     *                       the ItemStack object from.
+     * @return a deserialized ItemStack.
+     */
+    @Nullable
+    public static ItemStack deserialize(@Nonnull Map<String, Object> serializedItem) {
+        Objects.requireNonNull(serializedItem, "serializedItem cannot be null.");
+        return deserialize(mapToConfigSection(serializedItem));
+    }
+
+    @Nonnull
+    private static ConfigurationSection mapToConfigSection(@Nonnull Map<?, ?> serializedItem) {
```

Comment:
Is there any situation where `Map<String, Object>` is not available?

Reply:
Well, I don't know the internals of the deserialized map, maybe the deserialized map has `Integer` as keys at some point (because this method was designed to be called recursively, so the user might have some mapping from `Integer` to something at some nested level, or `int[]` as values, so I just went for the safe method, avoiding any `ClassCastException` that may arise there, plus it's not a big deal to use `Map<?, ?>` in that part of the code.